qualang_tools/config/components.py

```diff
@@ -646,20 +662,35 @@ def __init__(
 
         time_of_flight: Optional[int] = kwargs.get("time_of_flight", 0)
         smearing: Optional[int] = kwargs.get("smearing", 0)
-        digital_output_ports: Optional[List[DigitalInputPort]] = kwargs.get(
-            "digital_output_ports", None
-        )
         mixer: Optional[Mixer] = kwargs.get("mixer", None)
         pulses: Optional[List[Pulse]] = kwargs.get("pulses", None)
         operations: Optional[Dict[str, str]] = kwargs.get("operations", None)
+
+        analog_input_ports = _warn_if_not_none(kwargs, "analog_input_ports")
+        if analog_input_ports is not None:
+            element_analog_inputs = analog_input_ports
+
+        analog_output_ports = _warn_if_not_none(kwargs, "analog_output_ports")
+        if analog_output_ports is not None:
+            element_analog_outputs = analog_output_ports
+
+        digital_input_ports = _warn_if_not_none(kwargs, "digital_input_ports")
+        if digital_input_ports is not None:
+            element_digital_inputs = digital_input_ports
+
+        element_digital_outputs = []
+        digital_output_ports = _warn_if_not_none(kwargs, "digital_output_ports")
+        if digital_output_ports is not None:
+            element_digital_outputs = digital_output_ports
+
         super().__init__(
             name,
-            analog_input_ports,
-            analog_output_ports,
-            digital_input_ports,
-            digital_output_ports,
+            element_analog_inputs,
+            element_analog_outputs,
+            element_digital_inputs,
             intermediate_frequency,
             lo_frequency,
+            element_digital_outputs=element_digital_outputs,
```

Comment:
this line is a bit weird, what is it's purpose?

Reply:
we need this because, `element_digital_outputs` is not an argument of `Element`'s constructor

Reply:
Yes, now it all makes sense.